spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java

```diff
@@ -172,7 +172,11 @@ protected void assertEquals(String context, Object[] expectedRow, Object[] actua
       Object actualValue = actualRow[col];
       if (expectedValue != null && expectedValue.getClass().isArray()) {
         String newContext = String.format("%s (nested col %d)", context, col + 1);
-        assertEquals(newContext, (Object[]) expectedValue, (Object[]) actualValue);
+        if (expectedValue instanceof byte[]) {
+          Assert.assertArrayEquals(newContext, (byte[]) expectedValue, (byte[]) actualValue);
```

Comment:
`Assertions.assertThat(actualValue).as(newContext).isEqualTo(expectedValue)` is usually the better option than `assertArrayEquals`

Reply:
Thanks @nastra, just updated.